templates/ecs.yaml

```diff
@@ -0,0 +1,458 @@
+      - t3a.large
+    ConstraintDescription: Please choose a valid instance type.
+  DockerImageTag:
+    Type: String
+    Description: The Docker image to run from registry registry.gitlab.com/cri-iatlas/iatlas-api
+  # See https://docs.aws.amazon.com/AmazonECS/latest/developerguide/retrieve-ecs-optimized_AMI.html
+  LatestEcsAmiId:
+    Type: 'AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>'
+    Default: '/aws/service/ecs/optimized-ami/amazon-linux-2/recommended/image_id'
+    Description: A reference to the latest ECS Optimized AMI Id.
+  FlaskEnv:
+    Type: String
+    Description: The environment that the Flask application is running in. Valid values are `production`, `staging`, `development`, or `test`.
+    Default: 'production'
+    AllowedValues:
+      - test
+      - development
+      - staging
+      - production
+  DbHost:
+    Type: String
+    Description: The reader endpoint for the DB cluster. For example - mystack-mydbcluster-ro-123456789012.us-east-2.rds.amazonaws.com
+    NoEcho: true
+  DbPort:
+    Type: Number
+    Description: The port of the DB cluster.
+    Default: 5432
+    NoEcho: true
+  DbAccessSecurityGroup:
+    Type: String
+    Description: The DB Security Group.
+  DbCredsArn:
+    Description: The ARN of the Database secret in the Secret Manager
+    NoEcho: 'true'
+    Type: String
+  KmsKeyArn:
+    Description: 'Encryption key ARN to use for storage'
+    Type: String
+Resources:
+  ########################
+  ### Managed Policies ###
+  ########################
+  # This is necessary so it can read the Secrets Manager entry that contains the external registry credentials.
+  AccessSecretPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Action:
+              - kms:Decrypt
+              - secretsmanager:GetSecretValue
+              - ssm:GetParameters
+            Resource:
+              - !Ref 'KmsKeyArn'
+              - !Ref 'DbCredsArn'
+              - !Sub 'arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:iatlas_gitlab_registry_creds-??????'
+              - !Sub 'arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/*'
+          - Effect: Allow
+            Action:
+              - logs:CreateLogStream
+            Resource: '*'
+  EcsServicePolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Action:
+              - 'elasticloadbalancing:DeregisterInstancesFromLoadBalancer'
+              - 'elasticloadbalancing:DeregisterTargets'
+              - 'elasticloadbalancing:Describe*'
+              - 'elasticloadbalancing:RegisterInstancesWithLoadBalancer'
+              - 'elasticloadbalancing:RegisterTargets'
+              - 'ec2:Describe*'
+              - 'ec2:AuthorizeSecurityGroupIngress'
+            Resource: '*'
```

Comment:
shouldn't this be scoped to the `EcsAlb`?

Reply:
@zaro0508 - When I scoped this to the `EcsAlb`, it fails.